src/validation.cpp

```diff
@@ -465,6 +463,114 @@ static bool CheckInputsFromMempoolAndCache(const CTransaction& tx, TxValidationS
     return CheckInputScripts(tx, state, view, flags, /* cacheSigStore = */ true, /* cacheFullSciptStore = */ true, txdata);
 }
 
+/** An empty coin used as a placeholder for a spent coin.*/
+static const Coin coin_spent;
```

Comment:
I think this is only needed for spentness evaluation in `GetCoin` ? If so just make it part of the new class.